src/Nethermind/Nethermind.Blockchain/Receipts/PersistentReceiptStorage.cs

```diff
@@ -72,14 +72,14 @@ public class PersistentReceiptStorage : IReceiptStorage
 
         private void BlockTreeOnBlockAddedToMain(object? sender, BlockReplacementEventArgs e)
         {
-            if (e.PreviousBlock != null)
-            {
-                RemoveBlockTx(e.PreviousBlock);
-            }
-
             // Dont block main loop
             Task.Run(() =>
             {
+                if (e.PreviousBlock != null)
```

Comment:
No, can't do this here. There is another class `ReceiptCanonicalityMonitor` which sets the tx index by calling EnsureCanonical. If this is run after that is run, and the two block have the same tx hash, then the the tx index will be removed when it should not be removed.